ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/ccd/repositories/CCDCaseApi.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.cmc.ccd.migration.ccd.repositories;
+
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+import uk.gov.hmcts.cmc.ccd.mapper.CaseMapper;
+import uk.gov.hmcts.cmc.ccd.migration.idam.models.User;
+import uk.gov.hmcts.cmc.ccd.migration.idam.services.UserService;
+import uk.gov.hmcts.cmc.ccd.migration.models.mappers.JsonMapper;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
+public class CCDCaseApi {
+
+    public static final String JURISDICTION_ID = "CMC";
+    public static final String CASE_TYPE_ID = "MoneyClaimCase";
+
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final UserService userService;
+    private final CaseMapper caseMapper;
+    private final JsonMapper jsonMapper;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CCDCaseApi.class);
+
+    public CCDCaseApi(
+        CoreCaseDataApi coreCaseDataApi,
+        AuthTokenGenerator authTokenGenerator,
+        UserService userService,
+        CaseMapper caseMapper,
+        JsonMapper jsonMapper
+    ) {
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+        this.jsonMapper = jsonMapper;
+    }
+
+    public Optional<Map<String, Object>> getByReferenceNumber(String authorisation, String referenceNumber) {
+        LOGGER.info("Get claim from CCD " + referenceNumber);
+        User user = userService.getUser(authorisation);
+
+        List<Map<String, Object>> claims = extractClaims(
+            search(user, ImmutableMap.of("case.referenceNumber", referenceNumber))
+        );
+
+        if (claims.size() > 1) {
+            throw new RuntimeException("More than one claim found by claim reference " + referenceNumber);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS00112)

Reply:
ok